### DISCUSSION TITLE
flashing the SenseCap D1 (seed studio)

### DISCUSSION THREAD

**Discussion by ropper01** [HASwitchPlate/openHASP#557] [Q&A] [answered]
hi all, 

I want to flash my D1 but can't seem to find the firmware for it.. according to the hardware page it should be supported ? https://www.openhasp.com/0.7.0/hardware/seeed-studio/sensecap-indicator-d1/

many thanks

**Comment by fvanroie** [comment]
Hi,

It can be found in the asset list of the latest build under the [Github Actions tab](https://github.com/HASwitchPlate/openHASP/actions).
Click on the last successful build, scroll down to the artifacts and click to download the seeed-studios.zip file.

**Comment by fvanroie** [comment, marked as answer]
The SenseCap Indicator D1 frimware has been added to https://nightly.openhasp.com/

**Reply by hco** [reply]
I always get the message "Your ESP32-S3 board is not supported." on there… Is that expected?

**Reply by fvanroie** [reply]
Thank you for reporting this issue, the problem with the web installer has now been fixed.

**Reply by tome88** [reply]
Hi there, I'm getting a "Failed to initialize. Try resetting your device or holding the BOOT button while clicking INSTALL." when trying to flash my sensecap D1 using openhasp nightly installer. I've tried downloading firmware and flash using esphome flashing tool, via terminal and esptool.py but all getting timeout error messages. Also tried while holding down the bottom boot button on sensecap with no success

**Reply by ebo33** [reply]
I managed it with the Flash Download Tools
https://www.espressif.com/en/support/download/other-tools
As described here: https://www.openhasp.com/0.7.0/firmware/esp32/
![image](https://github.com/HASwitchPlate/openHASP/assets/171707031/3274a2eb-9923-42c9-a59d-626d91bb18c3)

**Reply by tome88** [reply]
> I managed it with the Flash Download Tools https://www.espressif.com/en/support/download/other-tools As described here: https://www.openhasp.com/0.7.0/firmware/esp32/ ![image](https://private-user-images.githubusercontent.com/171707031/336510643-3274a2eb-9923-42c9-a59d-626d91bb18c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NDg4OTMsIm5iZiI6MTcxNzY0ODU5MywicGF0aCI6Ii8xNzE3MDcwMzEvMzM2NTEwNjQzLTMyNzRhMmViLTk5MjMtNDJjOS1hNTlkLTYyNmQ5MWJiMThjMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNlQwNDM2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iN2UyMjc4ZGJhZjgyYzdjZGJkYmFhMmE1NDE3YzljZWJiMDc3MDk4MTU3Y2Y1NGM0MTc0ODIwNTJhOTZjY2U0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.m8n6tiDFmLVfu2DhmJaHZUwXPQjwY1Z99zTPOWwCPgc)

Thanks @ebo33 however when trying to flash it using the windows tool I get the error message:
`2-sync fail. Tips: Please refer to the chip dataset and confirm that strapping pin's status is correct.`
Any thoughts on how I'd address this? I've picked ESP32-S3 obviously as the board also.

**Reply by ebo33** [reply]
Just got a new SenseCap D1. Here are the screen shots from my flashing
![SenseCAP Frimware flashing1](https://github.com/user-attachments/assets/0d4b0508-5f72-49e5-8b57-af23c612622b)
![SenseCAP Frimware flashing2](https://github.com/user-attachments/assets/f5676cb9-8ed0-4910-a4de-353381e9f645)
![SenseCAP Frimware flashing3](https://github.com/user-attachments/assets/95823ba1-398c-47ee-96fc-0a72f9127cd6)
Use the Serial Port named "USB-SERIAL CH340" in device manager

**Reply by Jaxom99** [reply]
sooo, @ebo33 @tome88 did you actually managed to flash it ? 

I have the same trouble of Serial port busy when using the nightly webinstaller. I'll keep digging!

**Reply by Jaxom99** [reply]
Turns out another software `brltty` was hogging the ttyUSB0 port : so the CH340 connected, and was kicked 1 sec later. I found it through `dmesg`.

```
[119301.095472] usb 1-9.1: New USB device found, idVendor=1a86, idProduct=7523, bcdDevice= 2.64
[119301.095492] usb 1-9.1: New USB device strings: Mfr=0, Product=2, SerialNumber=0
[119301.095501] usb 1-9.1: Product: USB Serial
[119301.103858] ch341 1-9.1:1.0: ch341-uart converter detected
[119301.105257] usb 1-9.1: ch341-uart converter now attached to ttyUSB0
[119301.282783] usb 1-9.2: new full-speed USB device number 104 using xhci_hcd
[119301.882815] usb 1-9.2: device descriptor read/64, error -32
[119302.191024] usb 1-9.2: New USB device found, idVendor=2886, idProduct=0050, bcdDevice= 1.00
[119302.191047] usb 1-9.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[119302.191058] usb 1-9.2: Product: INDICATOR RP2040
[119302.191067] usb 1-9.2: Manufacturer: Seeed
[119302.191075] usb 1-9.2: SerialNumber: 4250304B38353816
[119302.205368] cdc_acm 1-9.2:1.0: ttyACM1: USB ACM device
[119303.119369] input: BRLTTY 6.6 Linux Screen Driver Keyboard as /devices/virtual/input/input72
[119303.128882] usb 1-9.1: usbfs: interface 0 claimed by ch341 while 'brltty' sets config #1
[119303.131071] ch341-uart ttyUSB0: ch341-uart converter now disconnected from ttyUSB0
[119303.131210] ch341 1-9.1:1.0: device disconnected
```

Once this program is disabled, ttyUSB0 stays up, and I can flash the SenseCAP alright !
